llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td

```diff
@@ -4268,17 +4269,16 @@ let Predicates = [HasSVE2p2orSME2p2] in {
   defm FCVT_ZPzZ : sve_fp_z2op_p_zd_b_0<"fcvt", "int_aarch64_sve_fcvt">;
 
   // SVE2p2 floating-point convert precision down (placing odd), zeroing predicate
-  defm FCVTNT_ZPzZ      : sve_fp_fcvtntz<"fcvtnt">;
-  def FCVTXNT_ZPzZ_DtoS : sve_fp_fcvt2z<0b0010, "fcvtxnt", ZPR32, ZPR64>;
+  defm FCVTNT_ZPzZ : sve2_fp_convert_down_narrow_z<"fcvtnt">;
+  def FCVTXNT_ZPzZ : sve2_fp_convert_precision<0b0010, 0b0, "fcvtxnt", ZPR32, ZPR64, /*destructive*/ true>;
```

Comment:
I believe we need tests with movprfx with FCVTNT and FCVTXNT.

Reply:
This patch hasn't changed whether `FCVTNT`/`FCVTXNT` can be movprfx'd - IIUC, the existing `movprfx` tests in in `MC/AArch64/SVE2p2/fcvtxnt_z-diagnostics.s` and `MC/AArch64/SVE2p2/fcvtnt_z-diagnostics.s` should not need to be changed?

Reply:
Sorry for my confusion, I saw destructive and for some reason I've mixed with:
let DestructiveInstType = Destructive<smth>;
Maybe we should change the name to haveconstrains, or something like that.

Reply:
I'm not sure. These instructions technically are destructive, they just can't be preceded by `movprfx` - see this from the Arn ARM

>A destructive instruction encoding is an instruction encoding where one of the source registers is also used as the
destination register.

I think the real issue is that setting `DestructiveInstType` to anything other than `NotDestructive` will allow the instructions to be preceded by a `movprfx`. I'd prefer that we refactor `DestructiveInstType`, but it be best done in a different patch.

What do you think?

Reply:
(1) Or one can copy instruction class instead of adding conditional dag patterns.

(2) Also one can go with decoupling the (currently implied) `movprfx`-ability property from the destructive property by adding an explicit flag `bool IsMOVPRFXAble = false` to `AArch64Inst` and setting it to true in the appropriate instructions.

(3) Alternatively, to save some typing, an uglier hackish way could be to  add a `movprfxz`-override flag and change the test to 
```
    if ((MCID.TSFlags & AArch64::NoMOVPRFX) || (MCID.TSFlags & AArch64::DestructiveInstTypeMask) ==
        AArch64::NotDestructive) { ...
```

I don't think (2) or (3) need to be in this patch though.
I would prefer (1), but not hugely important.

Reply:
Agree on the decoupling, but it does not really relate to (1) - no use of `sve_fp_convert_percision` is `movprfx`-able.

(1) would mean splitting `sve2_fp_convert_precision` into three classes; one for merging instructions, one for destructive zeroing instructions, and one for constructive zeroing instructions. I don't mind doing that, though I avoided it to reduce repetition; this file has 11 thousand lines at the time of writing.

Reply:
Ok, Fair enough.